### PR TITLE
Using unique log file for each workspace

### LINKED ISSUE #64: LS uses a global log file
I created a new user on Fedora 32 When I install the extension and open main.cf of a clean project is get the errors show in the screenshot:

Screenshot from 2020-11-13 11-26-52

The root cause is that this machine also has another user using LS. The install / start fails because the log file is owned by the other user and is private. What is also confusing is that it says "should i install" however, it is installed, it just does not start.

### PR DESCRIPTION
**Pull request by edvgui**
In this MR:
 - LS Log file has now a randomized path, and check that it is not used before choosing it.
 - Logging in LS has been improved
 - Logging in Extension has been extended

Closes #64

### CODE REVIEW

**Review by sanderr — APPROVED** [review]

**Review by sanderr — CHANGES_REQUESTED** [review]
Misclicked, forgot I had made comments.

File: .ci/Jenkinsfile-tests [hunk 1]
```diff
@@ -35,7 +36,7 @@ pipeline {
                 sh '''
                     rm -rf node_modules
                     npm i --also=dev
-                    INMANTA_LANGUAGE_SERVER_PATH="$INMANTA_LS_PATH" xvfb-run npm run test
+                    LOG_PATH="$INMANTA_LS_LOG_PATH" INMANTA_LANGUAGE_SERVER_PATH="$INMANTA_LS_PATH" xvfb-run npm run test
```
[the remark this answers is not in the document]
Comment (edvgui) [reply .ci/Jenkinsfile-tests]: Did it in https://github.com/inmanta/vscode-inmanta/pull/67/commits/9081bfd3a2a0faa35a29d9514bc69ac76436d234

File: server/src/inmantals/jsonrpc.py [hunk 2]
```diff
@@ -16,19 +16,36 @@
...
+def generate_safe_log_file():
+    # This env variable will only be set for testing purpose
+    file_name = os.getenv("LOG_PATH", None)
+    if file_name is not None:
+        return file_name
+
+    import random
+    import tempfile
+
+    file_name = "vscode-inmanta-%08x.log" % random.getrandbits(64)
```
[the remark this answers is not in the document]
Comment (edvgui) [reply server/src/inmantals/jsonrpc.py]: Did you mean something like that? https://github.com/inmanta/vscode-inmanta/pull/67/commits/8f24d5e569ec8c9aadeacbe38a596cd2e0e818b5 (should be ordered by name)

Or something more obvious (that would contain the hour, minutes, seconds more explicitly)?

[the remark this answers is not in the document]
Comment (sanderr) [reply server/src/inmantals/jsonrpc.py]: What you did is what I had in mind but either is fine for me.

**Review by sanderr — APPROVED** [review]

**Review by andraskvr — APPROVED** [review]

**Pull request merged**